onnxruntime/core/graph/ep_api_types.cc

```diff
@@ -538,23 +544,26 @@ Status EpGraph::Create(const GraphViewer& graph_viewer, /*out*/ std::unique_ptr<
 
       EpValueInfo* outer_value_info = value_info_iter->second.get();
       bool is_constant = false;
+      auto initializer_value = std::make_unique<OrtValue>();
       const ONNX_NAMESPACE::TensorProto* outer_initializer = parent_graph->GetInitializer(implicit_name,
```

Comment:
Do we need to heap allocate the OrtValue instance and use unique_ptr, or could we just use OrtValue directly given it contains a shared_ptr for the data?

Reply:
We could store them by value. The C API returns a pointer to an OrtValue. We could store them by value in an array (e.g., `std::vector<OrtValue> outer_values`) and then return their address (e.g., `&outer_values[i]`). I didn't default to this because a std::vector is reallocated as it grows, and if we ever add modification capabilities to `OrtGraph`, any pointers we handed to the user would be silently invalidated.

This is not a problem now since `OrtGraph` is read-only, but I supposed I'm being overly conservative. 

I'll update to store OrtValue by value and add a comment that mentions this possibility.

Reply:
Updated

Reply:
Updated.